samples/ChangefeedMigrationSample/Program.cs

```diff
@@ -0,0 +1,222 @@
+            Console.ReadKey();
+        }
+
+        async void RunChangeFeedProcessor()
+        {
+            // connect client 
+            DocumentClient client = new DocumentClient(new Uri(uri), this.secretKey);
+            await client.CreateDatabaseIfNotExistsAsync(new Database { Id = this.dbName });
+
+            // create monitor collection if it does not exist 
+            // WARNING: CreateDocumentCollectionIfNotExistsAsync will create a new 
+            // with reserved through pul which has pricing implications. For details
+            // visit: https://azure.microsoft.com/en-us/pricing/details/cosmos-db/
+            await client.CreateDocumentCollectionIfNotExistsAsync(
+                UriFactory.CreateDatabaseUri(this.dbName),
+                new DocumentCollection { Id = this.monitoredCollectionName },
+                new RequestOptions { OfferThroughput = 400 });
+
+            // create lease collect if it does not exist
+            // WARNING: CreateDocumentCollectionIfNotExistsAsync will create a new 
+            // with reserved through pul which has pricing implications. For details
+            // visit: https://azure.microsoft.com/en-us/pricing/details/cosmos-db/
+            await client.CreateDocumentCollectionIfNotExistsAsync(
+            UriFactory.CreateDatabaseUri(this.dbName),
+            new DocumentCollection { Id = this.leaseCollectionName },
+            new RequestOptions { OfferThroughput = 400 });
+            await StartChangeFeedHost();
+        }
+
+        async Task StartChangeFeedHost()
+        {
+            string hostName = Guid.NewGuid().ToString();
+
+            // monitored collection info 
+            DocumentCollectionInfo documentCollectionLocation = new DocumentCollectionInfo
+            {
+                Uri = new Uri(this.uri),
+                MasterKey = this.secretKey,
+                DatabaseName = this.dbName,
+                CollectionName = this.monitoredCollectionName
+            };
+
+            // lease collection info 
+            DocumentCollectionInfo leaseCollectionLocation = new DocumentCollectionInfo
+            {
+                Uri = new Uri(this.uri),
+                MasterKey = this.secretKey,
+                DatabaseName = this.dbName,
+                CollectionName = this.leaseCollectionName
+            };
+
+            ChangeFeedEventHost host = new ChangeFeedEventHost(hostName, documentCollectionLocation, leaseCollectionLocation);
+            await host.RegisterObserverAsync<DocumentFeedObserver>();
+            Console.WriteLine("Main program: press Enter to stop...");
+            Console.ReadLine();
+            await host.UnregisterObserversAsync();
+        }
+
+        private async Task UpdateDb()
+        {
+            // Use this function to update data in monitored collection in seperate thread
+            // Returns all documents in the collection.
+            Console.WriteLine("Connect client");
```

Comment:
It's not clear who is doing connect client -- change feed or update logic. Same for all messages below.